modules/id5IdSystem.js

```diff
@@ -129,39 +135,112 @@ export const id5IdSubmodule = {
    * @return {(IdResponse|function(callback:function))} A response object that contains id and/or callback.
    */
   extendId(config, cacheIdObj) {
-    const configParams = (config && config.params) || {};
-    incrementNb(configParams);
+    const partnerId = (config && config.params && config.params.partner) || 0;
+    incrementNb(partnerId);
     return cacheIdObj;
   }
 };
 
-function hasRequiredParams(configParams) {
-  if (!configParams || typeof configParams.partner !== 'number') {
+function hasRequiredConfig(config) {
+  if (!config || !config.params || !config.params.partner || typeof config.params.partner !== 'number') {
     utils.logError(`User ID - ID5 submodule requires partner to be defined as a number`);
     return false;
   }
+
+  if (!config || !config.storage || !config.storage.type || !config.storage.name) {
+    utils.logError(`User ID - ID5 submodule requires storage to be set`);
+    return false;
+  }
+
+  // TODO: in a future release, return false if storage type or name are not set as required
+  if (config.storage.type !== LOCAL_STORAGE) {
+    utils.logWarn(`User ID - ID5 submodule recommends storage type to be '${LOCAL_STORAGE}'. In a future release this will become a strict requirement`);
+  }
+  // TODO: in a future release, return false if storage type or name are not set as required
+  if (config.storage.name !== ID5_STORAGE_NAME) {
+    utils.logWarn(`User ID - ID5 submodule recommends storage name to be '${ID5_STORAGE_NAME}'. In a future release this will become a strict requirement`);
+  }
+
   return true;
 }
-function nbCookieName(configParams) {
-  return hasRequiredParams(configParams) ? `${BASE_NB_COOKIE_NAME}_${configParams.partner}_nb` : undefined;
+
+export function expDaysStr(expDays) {
+  return (new Date(Date.now() + (60 * 60 * 24 * expDays))).toUTCString();
```

Comment:
You probably don't want to multiply `60 * 60 * 24` with `expDays`?  
Your variable `NB_EXP_DAYS` already has this multiplication.  
I tested this on my local and I got `id5id_173_nb_exp: "Tue, 19 Jun 9117 08:02:10 GMT"`, which is a lot more than 30 days.

Reply:
ahh good catch, my bad. fixed